--- mobilesdk/linux/2.1.3.GA/module/android/bootstrap.py
+++ mobilesdk/linux/2.1.3.GA/module/android/bootstrap.py
@@ -332,7 +332,7 @@
 		open(genBindings, "w").write(gperfTemplate % gperfContext)
 
 	def generateNative(self, javaTemplate, cppTemplate, javaDir, cppDir):
-		jniPackage = self.moduleId.replace(".", "_")
+		jniPackage = self.moduleId.replace("_", "_1").replace(".", "_")
 		className = self.moduleName[0:1].upper() + self.moduleName[1:]
 
 		context = {

